### ISSUE TITLE
Add String.wrap and String.unwrap?

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#1237]
Inserts or removes `\n`'s.

**Comment by magnus-madsen** [comment]
Before you write any code, give a quick description of how you intend to implement it. For inspiration try to take a look at other librarys (e.g. lodash, underscore from JavaScript) or what else you can find :)

The idea is to discuss an existing implementation before we try to implement it in Flix.

**Comment by magnus-madsen** [comment]
This would still be nice to have... eventually :)

**Comment by sockmaster27** [comment]
Wrap as in 
```
wrap(11, "Lorem ipsum dolor sit ametametametamet")

      ↓

 Lorem ipsum
 dolor sit
 ametametam-
 etamet
|___________|
     11
```
and unwrapping like
```
unwrap("Lorem\nipsum")

      ↓

 Lorem ipsum
```
?

**Comment by magnus-madsen** [comment]
I think so, @mlutze ?

**Comment by magnus-madsen** [comment]
What does Haskell or other languages do?

**Comment by mlutze** [comment]
I'm not sure the hyphenation is possible though. Typically that happens around syllable boundaries.

**Comment by sockmaster27** [comment]
@mlutze 
Yes it would probably be too hard to do 100% correctly, but I was just thinking in the case of an overflowing word which couldn't be shifted in any other way. To avoid
```
wrap(11, "Lorem ipsum dolor sit ametametametamet")

      ↓

 Lorem ipsum
 dolor sit
 ametametametamet
|___________|
     11
```

**Comment by mlutze** [comment]
I think it's a fair design question.

Looking at a couple relevant libraries:

- [Apache WordUtils](https://commons.apache.org/proper/commons-text/javadocs/api-release/org/apache/commons/text/WordUtils.html#wrap(java.lang.String,int,java.lang.String,boolean)): does not wrap long words by default; but you can do it with an optional parameter
- [Text.Wrap](https://hackage.haskell.org/package/word-wrap-0.5/docs/Text-Wrap.html): takes a `WrapSettings` parameter; does not wrap long words by default


From this small sample it seems like wrapping long words is the less common default, but that it is desirable to have the option.

**Comment by magnus-madsen** [comment]
What's the simplest to implement? Let us start with that.

**Comment by magnus-madsen** [comment]
How difficult would it be to do the following:

If we wrap at say 80 chars and there is a word there then we move it to the next line. If there is a word which is more than 80 chars itself then we hardwrap it (i.e. break it in the middle)? Can that be done in say 20 lines of code?
